Entities/SLTerrain.cpp

```diff
@@ -1,1436 +1,474 @@
-
-    for (testY = 0; testY < pTempBitmap->h; ++testY)
-    {
-        for (testX = 0; testX < pTempBitmap->w; ++testX)
-        {
-            terrX = pos.m_X - halfWidth + testX;
-            terrY = pos.m_Y - halfHeight + testY;
-
-            // Make sure we're checking within bounds
-            if (terrX < 0) {
-                if (m_WrapX)
-                    while (terrX < 0)
-                        terrX += m_pMainBitmap->w;
-                else
-                    continue;
-            }
-            if (terrY < 0) {
-                if (m_WrapY)
-                    while (terrY < 0)
-                        terrY += m_pMainBitmap->h;
-                else
-                    continue;
-            }
-            if (terrX >= m_pMainBitmap->w) {
-                if (m_WrapX)
-                    terrX %= m_pMainBitmap->w;
-                else
-                    continue;
-            }
-            if (terrY >= m_pMainBitmap->h) {
-                if (m_WrapY)
-                    terrY %= m_pMainBitmap->h;
-                else
-                    continue;
-            }
-
-            testPixel = getpixel(pTempBitmap, testX, testY);
-            matPixel = getpixel(m_pMainBitmap, terrX, terrY);
-            colorPixel = getpixel(m_pFGColor->GetBitmap(), terrX, terrY);
-
-            if (testPixel != g_MaskColor)
-            {
-                // Only add PixelMO if we're not due to skip any
-                if (makeMOPs && matPixel != g_MaterialAir && colorPixel != g_MaskColor && ++skipCount > skipMOP && MOPDeque.size() < maxMOPs)
-                {
-                    skipCount = 0;
-                    sceneMat = g_SceneMan.GetMaterialFromID(matPixel);
-                    spawnMat = sceneMat->GetSpawnMaterial() ? g_SceneMan.GetMaterialFromID(sceneMat->GetSpawnMaterial()) : sceneMat;
-                    // Create the MOPixel based off the Terrain data.
-                    pPixel = new MOPixel(colorPixel,
-                                         spawnMat->GetPixelDensity(),
-                                         Vector(terrX, terrY),
-                                         Vector(),
-                                         new Atom(Vector(), spawnMat->GetIndex(), 0, colorPixel, 2),
-                                         0);
-
-                    pPixel->SetToHitMOs(false);
-                    MOPDeque.push_back(pPixel);
-                    pPixel = 0;
-                }
-
-                // Clear the terrain pixels
-                if (matPixel != g_MaterialAir)
-                    putpixel(m_pMainBitmap, terrX, terrY, g_MaterialAir);
-				if (colorPixel != g_MaskColor)
-				{
-					putpixel(m_pFGColor->GetBitmap(), terrX, terrY, g_MaskColor);
-					g_SceneMan.RegisterTerrainChange(terrX, terrY, 1, 1, g_MaskColor, false);
+	int SLTerrain::Save(Writer &writer) const {
+		SceneLayer::Save(writer);
+
+		// Only write the background texture info if the background itself is not saved out as a file already.
```

Comment:
I don't know how to reword this in a way that makes sense.
Basically you don't need the bg texture if you're providing a path to a pre-rendered bitmap because you won't be doing any texturization.
Same for the procgen stuff in FG, it will all be skipped.

Reply:
Okay, wanna chuck that bit of info into the comment?

I.e.
`// Only write the background texture info if the background itself is not saved out as a file already, since saved, pre-rendered bitmaps don't need texturing.`